sycl/include/sycl/device.hpp

```diff
@@ -214,8 +217,37 @@ class __SYCL_EXPORT device : public detail::OwnerLessBase<device> {
   /// type associated with the param parameter.
   ///
   /// \return device info of type described in Table 4.20.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  template <typename Param>
+  typename detail::is_device_info_desc<Param>::return_type get_info() const {
+    // For C++11_ABI compatibility, we handle these string Param types
+    // separately.
+    if constexpr (std::is_same_v<Param, info::device::name> ||
+                  std::is_same_v<Param, info::device::vendor> ||
+                  std::is_same_v<Param, info::device::driver_version> ||
+                  std::is_same_v<Param, info::device::version> ||
+                  std::is_same_v<Param, info::device::profile> ||
+                  std::is_same_v<Param, info::device::opencl_c_version> ||
+                  std::is_same_v<Param, info::device::ext_intel_pci_address>) {
+      detail::string_view PropertyName(typeid(Param).name());
+      detail::string Info = get_device_info(PropertyName);
+      return Info.c_str();
+    } else if constexpr (std::is_same_v<Param, info::device::extensions>) {
+      // return value is std::vector<std::string>
+      detail::string_view PropertyName(typeid(Param).name());
```

Comment:
This line can go outside `if constexpr` as it's shared between its branches.

Reply:
Yes, they are common only between two branches, but there are much more cases that do not belong to both branches.
In that case, we should not construct `detail::string_view` since it does not need marshalling.